boringtun/src/device/integration_tests/mod.rs

```diff
@@ -108,8 +110,8 @@ mod tests {
             // The local endpoint port is the remote listen port
             conf.push_str(&format!("\nListenPort = {}", self.endpoint.port()));
             conf.push_str(&format!(
-                "\nPrivateKey = {}",
```

Comment:
What does this change do?

Reply:
Secrets can't be trivially exported with x25519_dalek. You can _move_ the secret out, and you can run Diffie-Hellman operations with it, but you can't just get a ref and display it raw.

Reply:
Looks like this is what's breaking the integration tests. `gen_wg_conf` is trying to generate a config file for a fake server. That server needs a private key.